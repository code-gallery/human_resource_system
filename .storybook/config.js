import { configure } from '@storybook/react'

const loadStories = () => {
  require('../src/stories')
}

configure(loadStories, module)
