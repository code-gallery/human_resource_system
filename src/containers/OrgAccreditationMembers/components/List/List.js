import React from 'react'
import PropTypes from 'prop-types'
import _isNil from 'lodash/isNil'
import ListItem from '../ListItem'
import 'containers/UserVerifications/style.css'

const List = (props) => {
  const { data } = props

  if (_isNil(data) || (data && data.length === 0)) {
    return (
      <div>
        There are no verified users for this accreditation yet.
      </div>
    )
  }

  return (
    <table className="VerificationList">
      <thead className="hidden-sm hidden-xs">
        <tr className="row">
          <th className="col-xs-12 col-md-2 VerificationList-org">User</th>
          <th className="col-xs-12 col-md-2">Date</th>
          <th className="col-xs-12 col-md-4">Time</th>
        </tr>
      </thead>
      <tbody>
        {
          data.map((item, index) => {
            return (
              <ListItem
                key={index}
                item={item}
              />
            )
          })
        }
      </tbody>
    </table>
  )
}

List.propTypes = {
  data: PropTypes.array
}

export default List
