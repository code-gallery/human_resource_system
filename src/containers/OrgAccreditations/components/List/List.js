import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ContentCard from 'components/ContentCard'
import LoadingIndicator from 'components/LoadingIndicator'
import ListItem from '../ListItem'

class List extends Component {
  renderTable() {
    const { awards, awardTypes, deleteAccreditation } = this.props
    return (
      <table className="VerificationList">
        <thead className="hidden-sm hidden-xs">
          <tr className="row">
            <th className="col-xs-12 col-md-2">Name</th>
            <th className="col-xs-12 col-md-2">Type</th>
            <th className="col-xs-12 col-md-2">Delivery Type</th>
            <th className="col-xs-12 col-md-2">Created</th>
            <th className="col-xs-12 col-md-2">Status</th>
            <th className="col-xs-12 col-md-2">Actions</th>
          </tr>
        </thead>
        <tbody>
          {
            awards.map((item, key) => {
              return (
                <ListItem
                  key={key}
                  item={item}
                  awardTypes={awardTypes}
                  deleteAccreditation={deleteAccreditation}
                />
              )
            })
          }
        </tbody>
      </table>
    )
  }

  render() {
    const { pending, awards } = this.props
    if (pending === 'fetching') {
      return (
        <ContentCard title="Current">
          <LoadingIndicator display="inline" />
        </ContentCard>
      )
    }

    return (
      <ContentCard title="Current">
        {awards.length === 0 &&
          <p>There are no accreditations at the moment</p>
        }
        {awards.length > 0 &&
          this.renderTable()
        }
      </ContentCard>
    )
  }
}

List.propTypes = {
  awards: PropTypes.array,
  awardTypes: PropTypes.object,
  deleteAccreditation: PropTypes.func.isRequired,
  pending: PropTypes.string
}

export default List
