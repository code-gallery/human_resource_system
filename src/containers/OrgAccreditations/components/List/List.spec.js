import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import config from 'store'
import List from './List'

describe('containers/OrgAccreditations/components/List', () => {
  const { store } = config
  describe('when props.pending is "fetching"', () => {
    it('renders correctly', () => {
      const props = {
        awards: [],
        deleteAccreditation: jest.fn(),
        pending: 'fetching'
      }
      const tree = renderer.create(
        <Provider store={store}>
          <Router>
            <List {...props} />
          </Router>
        </Provider>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.awards is empty', () => {
    it('renders correctly', () => {
      const props = {
        awards: [],
        deleteAccreditation: jest.fn(),
        pending: 'fetchSuccess'
      }
      const tree = renderer.create(
        <Provider store={store}>
          <Router>
            <List {...props} />
          </Router>
        </Provider>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.awards is not empty', () => {
    it('renders correctly', () => {
      const props = {
        awards: [
          {
            id: 1, name: 'Blockchain', level: 'Level 1',
            enabled: 1, delivery_type: 'In a classroom', type: 'cpd',
            organisation_id: 100
          },
          {
            id: 2, name: 'Award 1', level: 'Level 2',
            enabled: 0, delivery_type: 'Online', type: 'award',
            organisation_id: 200
          }
        ],
        awardTypes: {
          cpd: { label: 'CPD' },
          award: { label: 'Award' }
        },
        deleteAccreditation: jest.fn(),
        pending: 'fetchSuccess'
      }
      const tree = renderer.create(
        <Provider store={store}>
          <Router>
            <List {...props} />
          </Router>
        </Provider>
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
