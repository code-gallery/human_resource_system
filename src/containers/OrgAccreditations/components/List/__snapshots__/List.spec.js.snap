// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/OrgAccreditations/components/List when props.awards is empty renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 "
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Current
    </h3>
    <p>
      There are no accreditations at the moment
    </p>
  </div>
</div>
`;

exports[`containers/OrgAccreditations/components/List when props.awards is not empty renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 "
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Current
    </h3>
    <table
      className="VerificationList"
    >
      <thead
        className="hidden-sm hidden-xs"
      >
        <tr
          className="row"
        >
          <th
            className="col-xs-12 col-md-2"
          >
            Name
          </th>
          <th
            className="col-xs-12 col-md-2"
          >
            Type
          </th>
          <th
            className="col-xs-12 col-md-2"
          >
            Delivery Type
          </th>
          <th
            className="col-xs-12 col-md-2"
          >
            Created
          </th>
          <th
            className="col-xs-12 col-md-2"
          >
            Status
          </th>
          <th
            className="col-xs-12 col-md-2"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="row ListItem"
        >
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              Blockchain
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p />
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              In a classroom
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              April 21st 2018
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              Active
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p
              className="table-btn-holder"
            >
              <a
                className="btn yellow-btn action-btn"
                href="/organisations/100/accreditations/1/qrcode"
                onClick={[Function]}
              >
                QR code
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <a
                className="btn blue-btn action-btn text-center"
                href="/organisations/100/accreditations/1/edit"
                onClick={[Function]}
              >
                Edit
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <a
                className="btn green-btn action-btn text-center"
                href="/organisations/100/accreditations/1/members"
                onClick={[Function]}
              >
                Verified Users
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <button
                className="btn red-btn action-btn"
                onClick={[Function]}
              >
                Delete
              </button>
            </p>
          </td>
        </tr>
        <tr
          className="row ListItem"
        >
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              Award 1
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p />
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              Online
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              April 21st 2018
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p>
              Inactive
            </p>
          </td>
          <td
            className="col-xs-12 col-md-2"
          >
            <p
              className="table-btn-holder"
            >
              <a
                className="btn yellow-btn action-btn"
                href="/organisations/200/accreditations/2/qrcode"
                onClick={[Function]}
              >
                QR code
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <a
                className="btn blue-btn action-btn text-center"
                href="/organisations/200/accreditations/2/edit"
                onClick={[Function]}
              >
                Edit
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <a
                className="btn green-btn action-btn text-center"
                href="/organisations/200/accreditations/2/members"
                onClick={[Function]}
              >
                Verified Users
              </a>
            </p>
            <p
              className="table-btn-holder"
            >
              <button
                className="btn red-btn action-btn"
                onClick={[Function]}
              >
                Delete
              </button>
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`containers/OrgAccreditations/components/List when props.pending is "fetching" renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 "
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Current
    </h3>
    <div
      className="LoadingIndicator LoadingIndicator-inline"
    >
      <svg
        className=""
        height="40px"
        version="1.1"
        viewBox="0 0 50 50"
        width="40px"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
        y="0px"
      >
        <path
          d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
          fill="#FF6700"
        >
          <animateTransform
            attributeName="transform"
            attributeType="xml"
            dur="0.6s"
            from="0 25 25"
            repeatCount="indefinite"
            to="360 25 25"
            type="rotate"
          />
        </path>
      </svg>
    </div>
  </div>
</div>
`;
